lib/fs.js

```diff
@@ -1328,21 +1340,34 @@ FSWatcher.prototype.start = function(filename,
                                      persistent,
                                      recursive,
                                      encoding) {
+  lazyAssert()(this._handle instanceof FSEvent, 'handle must be a FSEvent');
+  if (this._handle.initialized) {
+    throw new errors.Error('ERR_FS_WATCHER_ALREADY_STARTED');
+  }
+
   filename = getPathFromURL(filename);
-  nullCheck(filename, 'filename');
+  validatePath(filename, 'filename');
+
   var err = this._handle.start(pathModule.toNamespacedPath(filename),
-                               persistent,
-                               recursive,
+                               !!persistent,
+                               !!recursive,
```

Comment:
Minor if correct change in behavior.

Right now, C++ land calls IsTrue(), which is essentially a strict `x === true` check.  Using coercion here changes the observable behavior to a non-strict `x == true` check.

Reply:
@bnoordhuis Yes, generally I think for boolean options it seems to be more intuitive to coerce them as if they are handled like `if (options.something)`. The default cases are set in `fs.watch` so we should be fine in the common cases.

Reply:
I personally think it is best to have a strict check. Why would we want to accept any thruthy value?